src/main/java/core/basesyntax/service/impl/OperationStrategyImpl.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.OperationStrategy;
+import core.basesyntax.strategy.OperationHandler;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    @Override
+    public OperationHandler getStrategy(
+            Map<String, OperationHandler> operationHandlerMap, String[] line) {
+        return operationHandlerMap.get(line[0]);
+    }
```

Comment:
```suggestion
    Map<String, OperationHandler> operationHandlers;

    public OperationStrategyImpl(
        Map<String, OperationHandler> operationHandlers) {
        this.operationHandlers = operationHandlers;
    }

    @Override
    public OperationHandler getStrategy(String operation) {
        return operationHandlers.get(operation);
    }
```

lets pass map through constructor

Reply:
can't get what is it for? What principally changes?

Reply:
did and got why, thanks!